README.md

```diff
@@ -993,6 +995,154 @@ thing you need to provide is the `user_type` you want to create.
+load test to cover an endpoint that will reject the request because of some error (eTag, bad data,
+etc.). If you want to do that, you can do something like this (it's a bit long because of setup):
+
+```python
+# -*- coding: utf-8 -*-
+"""
+Example of a TaskSet file...
+"""
+import json
+import logging
+import random
+from copy import deepcopy
+from http import HTTPStatus
+
+from locust import tag, task
+
+from utils.base import is_local
+from utils.constants import MTO_SHIPMENT
+from utils.parsers import APIKey, get_api_fake_data_generator
+from utils.request import log_response_failure, log_response_info
+from utils.rest import RestResponseContextManager
+from utils.task import RestTaskSet
+from utils.types import JSONArray, JSONObject
+
+
+logger = logging.getLogger(__name__)
+fake_data_generator = get_api_fake_data_generator()
+
+
+# tags are useful to add at a TaskSet and task level to enable running only specific load tests.
+@tag('mainTasks')
+class PrimeTasks(RestTaskSet):
+  """
+  Description of the flow for these tasks here.
+  """
+
+  @task
+  def stop(self) -> None:
+    """
+    This ensures that at some point, the user will stop running the tasks in this task set.
+    """
+    self.interrupt()
+
+  # You can add the expectedFailure tag so that they can all be run if needed.
+  @tag(MTO_SHIPMENT, "updateMTOShipmentStatus", "expectedFailure")
+  @task
+  def update_mto_shipment_with_invalid_status(self) -> None:
+    """
+    Tries updating an MTO shipment to an invalid status.
+    """
+    # Need a move to work with first. For the sake of this example, we'll get a random move that is
+    # available to prime.
+    url, request_kwargs = self.request_preparer.prep_prime_request(endpoint="/moves")
+
+    if is_local(env=self.env):
+      request_kwargs["timeout"] = 15  # set a timeout of 15sec if we're running locally for this endpoint
+
+    with self.rest(method="GET", url=url, **request_kwargs) as resp:
+      resp: RestResponseContextManager
+
+      log_response_info(response=resp)
+
+      if resp.status_code == HTTPStatus.OK:
+        moves: JSONArray = resp.js
+      else:
+        resp.failure("Unable to get moves available to prime.")
+
+        log_response_failure(response=resp)
+
+        return
+
+    move = random.choice(moves)
+
+    # Now we need to retrieve the shipments for this move
+    url, request_kwargs = self.request_preparer.prep_prime_request(
+      endpoint=f"/move-task-orders/{move['id']}",
+      endpoint_name="/move-task-orders/{moveID}",
+    )
+
+    with self.rest(method="GET", url=url, **request_kwargs) as resp:
+      resp: RestResponseContextManager
+
+      log_response_info(response=resp)
+
+      if resp.status_code == HTTPStatus.OK:
+        move: JSONObject = resp.js
+      else:
+        resp.failure("Unable to get the move.")
+
+        log_response_failure(response=resp)
+
+        return
```

Comment:
Because this an example, I think we should rely on just assuming there is a stored shipment we want to update to keep the focus on the failure request.  I think all of the setup detracts from the point we're trying to get across.

Reply:
I was torn on this because the storage as is right now causes several tests to fail so I wouldn't want someone copying this and then their test failing. I created https://dp3.atlassian.net/browse/MB-11102 to re-evaluate the storage. 

Though maybe since I ended up making this a real load test, I can shorten it in the readme and tell people to go look at the real one to see the full version.

Reply:
I'd probably lean towards a smaller scoped example so that its more focused and then linking out/referencing a specific example.

As a separate point, it might be worth adding something (we can probably add it to the story mentioned above) that in the current flow with local store we should likely be updating the store anytime a PUT, POST, or PATCH is successfully completed, so that when we try and make changes to the same object we have the updated eTags.

Reply:
I agree about the setup detracting from the point of the example so I'm cutting it down now.